tests/bofire/outlier_detection/test_outlier_detection.py

```diff
@@ -54,11 +54,11 @@ def test_IterativeTrimming():
         )
     )
     ITGP = mapper.map(ITGP_model)
-    assert isinstance(ITGP.base_gp, SingleTaskGPSurrogate)
+    assert isinstance(ITGP.base_gp, SingleTaskGPSurrogate)  # type: ignore
     assert isinstance(ITGP, mapper.IterativeTrimming)
     # detect
-    trimmed, outliers = ITGP.detect(experiments=experiments)
-    assert isinstance(trimmed, pd.DataFrame)
-    assert isinstance(outliers, pd.DataFrame)
-    assert len(experiments) == len(trimmed) + len(outliers)
-    assert len(outliers) >= n_outlier / 2
+    experiments1 = ITGP.detect(experiments=experiments)
+    assert len(experiments[experiments["valid_y"] == 1]) != len(
+        experiments1[experiments1["valid_y"] == 1]
+    )
+    assert len(experiments1[experiments1["valid_y"] == 0]) > n_outlier / 2
```

Comment:
Can you also add a test for the functional testing `OutlierDetections`? class to see if detection outliers for several outputs in parallel works?